src/api/meal_plan.py

```diff
@@ -1,89 +1,227 @@
-                        CAST(m.energy_amount AS double precision),
-                        u.energy_amount,
-                        un.energy_amount
-                    )
-                ),
-                0
-            ) AS total_calories
-            FROM meal_logs ml
-            JOIN recipe_ingredients ri ON ml.recipe_id = ri.recipe_id
-            LEFT JOIN menustat m ON ri.ingredient_id = m.menustat_id
-            LEFT JOIN usda_branded u ON ri.ingredient_id = u.fdc_id
-            LEFT JOIN usda_non_branded un ON ri.ingredient_id = un.id
-            WHERE ml.meal_plan_id = :meal_plan_id AND ml.date = :date
-        """), {"meal_plan_id": meal_plan_id, "date": date})
+        if author_id is recipient_id:
+            raise HTTPException(status_code=400, detail="Author id and recipient id can not be the same.")
+
+        try:
+            validate_meal_plan = connection.execute(text("""SELECT 1 FROM meal_plans WHERE id = :id AND author_id = :author_id"""), {"id": meal_plan_id, "author_id": author_id}).one_or_none()
+
+            if validate_meal_plan is None:
+                    raise Exception("Meal plan and author id combination not found.")
+            
+            connection.execute(text("""INSERT INTO shared_meal_plans (meal_plan_id, recipient_id) VALUES (:meal_plan_id, :recipient_id)"""), {"meal_plan_id": meal_plan_id, "recipient_id": recipient_id})
+            
+            return Response(status_code=200)
         
-        total_calories = result.fetchone()[0]
-    
-    return {"date": date, "total_calories": total_calories}
+        except Exception as e:
+            raise HTTPException(status_code=404, detail=f"Error sharing plan: {e}")
+        
+@router.get("/{account_id}")
+def get_meal_plans(account_id: int):
+    try:
+        with db.engine.begin() as connection:
+            validate_user = connection.execute(text("""SELECT 1 FROM accounts WHERE id = :account_id"""), {"account_id": account_id}).one_or_none()
+
+            if validate_user is None:
+                raise Exception("Account not found.")
+
+            results = connection.execute(text("""SELECT id, name, description
+                                            FROM meal_plans
+                                            WHERE author_id = :account_id 
+                                            UNION 
+                                            SELECT meal_plan_id, name, description FROM shared_meal_plans 
+                                            JOIN meal_plans ON shared_meal_plans.meal_plan_id = meal_plans.id 
+                                            WHERE recipient_id = :account_id"""), {"account_id": account_id}).mappings().all()
+        
+            if results:
+                return results
+            
+            return []
+
+    except Exception as e:
+        raise HTTPException(status_code=404, detail=f"{e}")
+
+@router.get("/{meal_plan_id}/{account_id}")
+def get_meal_plan(meal_plan_id: int, account_id: int):
```

Comment:
This becomes inconsistent with the endpoint above, as doing `/meal-plans/123/` queries for `user_id = 123`, but then doing `/meal-plans/123/1` now queries for `meal_plan.id = 123` and `user_id = 1`. If I was an end user, I would assume `user_id = 123` and `meal_plan.id = 1`.
I think it would make sense to change this to just `/meal-plans/{meal_plan_id}`, then include `user_id` in the request body

Reply:
I tried to do it like that because it made the most sense to me. However since the endpoint is ```meal-plan/{int}``` it is the same endpoint, so thats why I added the extra id at the end of the endpoint.

Reply:
If the first endpoint is moved to be under accounts, isn't that not an issue anymore?

Reply:
GREAT SCOTT YOU ARE RIGHT